inventory/src/main/java/com/yas/inventory/repository/StockRepository.java

```diff
@@ -16,4 +16,7 @@ List<Stock> findByWarehouseIdAndProductIdIn(Long warehouseId,
                                                 List<Long> productIds);
 
     boolean existsByWarehouseIdAndProductId(Long warehouseId, Long productId);
+
+    @Query("SELECT distinct s.productId FROM Stock s")
+    List<Long> findProductIdsAddedWarehouse();
```

Comment:
Do we have cases where quantity = 0? Those records should be excluded, right?
`findProductIdsInStock` may be a better name.